src/controller/agent.governance.ts

```diff
@@ -113,22 +119,43 @@ export class AgentGovernance {
         }
     }
 
-    // register a callback if to receive notification that a particular message
-    // has been received by governance policy.  Note:  currently only the basic message handler
-    // consumes invokeHandler (below).  TODO if we have more use cases, then we should move the invocation higher up
-    public registerHandler(topic: string, func: ControllerCallback) {
-        this.callbacks.push({topic, func, exceptionCount: 0});
+    /**
+     * register a callback if to receive notification that a particular message
+     * has been received by governance policy.  Note:  currently only the basic message handler
+     * consumes invokeHandler (below).  TODO if we have more use cases, then we should move the invocation higher up
+     *
+     * @param id a value that makes debugging easier. duplicates will not break the system
+     * @param topic a value matching values sent by acapy webhook topic parameter
+     * @param func the callback, must be of type ControllerCallback
+     */
+    public registerHandler(id: string, topic: string, func: ControllerCallback) {
+        this.callbacks.push({id, topic, func, exceptionCount: 0});
     }
 
+    /**
+     * Iterate through all of the handlers registered and call the ones that match for the topic
+     * This function returns true if 1 or more of the handlers processed the message.  It does not guarantee
+     * that only 1 handler processed the message or that all of the handlers processed the message.
+     *
+     */
     public async invokeHandler(agentUrl: string, agentId: string, adminApiKey: string, route: string, topic: string,
                                body: any, token?: string) : Promise<any> {
+        let result: boolean = false;
         // tslint:disable-next-line:forin
         for (const index in this.callbacks) {
             const registration: Registration = this.callbacks[index];
-            Logger.debug(`callback found ${registration.topic}`, registration);
+            Logger.debug(`callback found topic: ${registration.topic} id: ${registration.id}`);
             if (registration.topic === topic) {
-                await registration.func(agentUrl, agentId, adminApiKey, route, topic, body, token);
+                const funResult: boolean = await registration.func(agentUrl, agentId, adminApiKey, route, topic, body, token);
+                if (funResult === true) {
```

Comment:
Another option to reduce nesting would be to change this `if` block to `result = funResult || result`. Definitely not necessary though.